### PR TITLE
Travis for CI

### PR DESCRIPTION
**Pull request by BPScott**
Shopify Build now fails for public projects as shipping folks haven't yet worked out the risks and mitigations around letting randos run code on our infrastructure. That's reasonable.

We currently have a few builds on Shopify build: polaris-icons-production-builder and some other deployment related ones: https://buildkite.com/shopify?filter=polaris-icons

Proposed steps:

- [x] Use Travis for PRs for tests/linting (this PR)
- [x] Delete polaris-icons job (https://buildkite.com/shopify/polaris-icons and https://github.com/Shopify/polaris-icons/blob/master/.shopify-build/polaris-icons.yml) as it is now handled by Travis jobs. (this PR)
- [x] Keep the other pipelines on Shopify Build, but limit them to only run on the master branch. Thus only blessed (i.e. merged by a Shopify person) code will run on Shopify Build, mitigating risk of randos running arbitrary code on Shopify infrastructure.
- [x] Update shipit configs to point to statuses from Travis instead of removed buildkit contig (this PR)
- [ ] Ask for an exception for this repo, as Shopify Build pipelines will only run against the master branch, meaning code has been vetted and deemed safe by a Shopify employee as part of code review.

### Why do we want to keep `polaris-icons-production-builder` and other deployment based builds?

This monorepo contains an app that gets deployed on Shopify Cloud, `polaris-icons-production-builder`, the k8s validator and bin checker act as preflight checks to ensure the container can be built and deployed. I don't believe we'll be easily be able to set that up as a Travis build. Without this we will have no guarantee that our deployments will work. While having these checks on PRs is nice, having them on master only will be enough to ensure we don't deploy something broken to https://polaris-icons.shopify.com, while ensuring arbitrary untrusted code is not ran on Shopify infra.

Our deployment of rubygems requires a `polaris-icons-deploy-ruby` build to create the gem and push it into our packagecloud instance as this can't be done within shipit (as it doesn't have `librsvg2-bin` installed which is required to generate the icons to be placed into the gem). Travis can be setup to check if a gem can be build on PR, but pushing to packagecloud is more convoluted. Travis supports [packagecloud](https://docs.travis-ci.com/user/deployment/packagecloud/) but it is unclear if they support enterprise hosted instances. This build can be is only ran against the master branch when we want to release so the risk of running arbitrary untrusted code is mitigated.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  include:
+    - language: node_js
+      node_js:
+        - '10'
+      cache: yarn
+      script:
+        - yarn run build
+        - yarn run type-check
+        - yarn run test
+        - yarn run lint
```
Comment (amrocha) [inline .travis.yml]: Not sure how Travis sorts these, but it'd be nice to run the linting either as the first step or concurrently with the other steps so any errors are caught quickly

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    - language: ruby
+      rvm:
+        - '2.6.2'
+      before_install:
+        - sudo apt-get install -y librsvg2-bin
```
Comment (amrocha) [inline .travis.yml]: No need to update?

Comment (BPScott) [reply .travis.yml]: Not that I can tell https://travis-ci.com/Shopify/polaris-icons/jobs/218365254

File: .travis.yml (hunk 1 above)
Comment (BPScott) [reply .travis.yml]: It's the order of this array.

Build takes 20s (and must go before the tests and type-check), the rest take 10s each. Build scaffholding (setting up and tearing down the container, doing the `yarn install` take the remaining 60s of build time)

I don't think there's much value in spinning up an extra container to shave 10s off the build time.

**Pull request merged**